### DISCUSSION TITLE
x-bind:style mix with existing styles on element

### DISCUSSION THREAD

**Discussion by josephmiclaus** [alpinejs/alpine#2469] [1. Help] [answered]
Hi there!

I was experimenting around with binding styles to an element and stumbled upon the following issue.

Let's say we have a `card` component which has some a style attribute defining `width` and `height` and also a `background-image`. (The `background-image` is defined in `style` in order for it to already be displayed before Alpine loads.)

When the Alpine component initializes, there is a timeout, after which the card image is changed.

However when we do it this way `:styles` overrides the `style` attribute instead of mixing it with the exisiting styles defined there. So the defined `width` and `height` properties are removed after the `data.img` property is changed.

Is there a way to fix this, I'm not seeing, or should one simply duplicate the styles (in our case `width` and `height`) and also put them in `:styles`? I tried this and it works, but it would be nice to prevent some code duplication.

(Also could we use the object-based syntax for `background-image` like so `backgroundImage` instead?)

DOM:
```html
<div x-data="card"
    class="rounded-2xl flex items-center justify-center bg-cover bg-center text-white text-3xl"
    style="width: 640px; height: 480px; background-image: url(http://via.placeholder.com/640x480);"
    :style="`background-image: url(${data.img});`"
>
    <h1 class="pb-4">Tech</h1>
</div>
```

Alpine component:
```js
Alpine.data('card', () => ({
    data: {
        img: 'http://via.placeholder.com/640x480',
    },

    init() {
        setTimeout(() => {
            this.data.img = 'https://placeimg.com/640/480/tech';
        }, 3000);
    },
}));
```
---
Solution (however with code duplication):
```html
<div x-data="card"
    class="rounded-2xl flex items-center justify-center bg-cover bg-center text-white text-3xl"
    style="width: 640px; height: 480px; background-image: url(http://via.placeholder.com/640x480);"
    :style="`width: 640px; height: 480px; background-image: url(${data.img});`"
>
    <h1 class="pb-4">Tech</h1>
</div>
```
Thanks!

**Comment by SimoTod** [comment, marked as answer]
If you set a string directly, it will override the whole attribute. If you use an object, it will just override the specific property and keep the other ones as they are. See https://alpinejs.dev/directives/bind#binding-styles

**Reply by josephmiclaus** [reply]
Okay got it, thanks for the nudge!

Here's the code:
```html
<div x-data="card"
    class="rounded-2xl flex items-center justify-center bg-cover bg-center text-white text-3xl"
    style="width: 640px; height: 480px; background-image: url(http://via.placeholder.com/640x480);"
    :style="{ backgroundImage: 'url(' + data.img + ')' }"
>
    <h1 class="pb-4">Tech</h1>
</div>
```

Also possible:
```html
<div x-data="card"
    class="rounded-2xl flex items-center justify-center bg-cover bg-center text-white text-3xl"
    style="width: 640px; height: 480px; background-image: url(http://via.placeholder.com/640x480);"
    :style="{ backgroundImage: `url(${data.img})` }"
>
    <h1 class="pb-4">Tech</h1>
</div>
```
